src/state_machine/hashmap.rs

```diff
@@ -0,0 +1,62 @@
+#![cfg(feature="hashmap")]
+use std::io::{Error, Result};
+use std::collections::HashMap;
+
+use serde::json::{self, Value};
+
+use state_machine::StateMachine;
+
+/// A state machine that holds a single mutable value.
+#[derive(Debug)]
+pub struct HashmapStateMachine {
+    map: HashMap<String, Value>,
```

Comment:
Maybe I'm missing the point of this, but why are you constraining the key and value types?  It seems like both should be generic and bounded by `Serialize` and `Deserialize`.